sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -314,9 +314,9 @@ public void processElement(ProcessContext context) {
 
     @Override
     public void validate(PBegin input) {
-      checkNotNull(getQuery(), "query");
-      checkNotNull(getRowMapper(), "rowMapper");
-      checkNotNull(getCoder(), "coder");
+      checkNotNull(getQuery(), "Query should be set");
```

Comment:
Here, I would rather have something like "JdbcIO requires a query to be set via withQuery(...)" or "attempt to use JdbcIO prior to setting query with withQuery(...)", etc, so the user is directed to how to fix their code without having to interpret the stack trace.

Again, NPE almost always looks like a bug in our code. I think the right exception is `IllegalStateException` which you can get with `checkState(query != null, ...)`